### PR TITLE
[Lang] [bug] Fix numpy from and to ndarray matrix

### PR DESCRIPTION
**Pull request by qiao-bo**
Related issue = #

### CODE REVIEW

**Review by strongoier — APPROVED** [review]
LGTM!

File: python/taichi/lang/_ndarray.py [hunk 1]
```diff
@@ -89,34 +89,41 @@ def __getitem__(self, key):
...
 
         Returns:
             numpy.ndarray: The result numpy array.
         """
         if impl.current_cfg().ndarray_use_torch:
             return self.arr.cpu().numpy()
+
         arr = np.zeros(shape=self.arr.shape, dtype=to_numpy_type(self.dtype))
-        taichi.lang.meta.ndarray_to_ext_arr(self, arr)
+        if len(self.element_shape) == 0:
```
Comment (strongoier) [inline python/taichi/lang/_ndarray.py:115]: nit: avoid manually checking this in the parent class

**Review by ailzhang — APPROVED** [review]
Thanks for the quick fix!!

File: python/taichi/lang/_ndarray.py (hunk 1 above)
Comment (qiao-bo) [reply python/taichi/lang/_ndarray.py]: @strongoier fixed!

**Pull request merged**